File: packages/dd-trace/test/llmobs/parity/scenarios/anthropic/messages-stream.py
import os

from anthropic import Anthropic

from common import finish, start


start("anthropic")
client = Anthropic(api_key=os.environ["ANTHROPIC_API_KEY"], base_url=os.environ["PROVIDER_BASE_URL"])
stream = client.messages.create(
    model="claude-3-5-sonnet-20241022",
    max_tokens=16,
    stream=True,
    messages=[{"role": "user", "content": "Stream a short parity response."}],
)
for chunk in stream:
    del chunk
finish()

File: packages/dd-trace/test/llmobs/parity/scenarios/anthropic/messages-stream.js
'use strict'

const { finish, loadTracer, loadVersionedModule } = require('../common')

async function main () {
  const tracer = loadTracer('anthropic')
  const { Anthropic } = loadVersionedModule('@anthropic-ai/sdk')
  const client = new Anthropic({ apiKey: process.env.ANTHROPIC_API_KEY, baseURL: process.env.PROVIDER_BASE_URL })
  const stream = await client.messages.create({
    model: 'claude-3-5-sonnet-20241022',
    max_tokens: 16,
    stream: true,
    messages: [{ role: 'user', content: 'Stream a short parity response.' }],
  })
  for await (const chunk of stream) void chunk
  await finish(tracer)
}

main().catch(error => {
  process.stderr.write(`${error.stack ?? error}\n`)
  process.exitCode = 1
})
